R/dummy.R

```diff
@@ -98,35 +100,50 @@
 #' # Default dummy coding: 36 dummy variables
 #' dummies <- rec %>%
 #'   step_dummy(city) %>%
-#'   prep(training = Sacramento)
+#'   prep()
 #'
 #' dummy_data <- bake(dummies, new_data = NULL)
 #'
 #' dummy_data %>%
 #'   select(starts_with("city")) %>%
-#'   names() # level "anything" is the reference level
+#'   glimpse() # level "anything" is the reference level
 #'
 #' # Obtain the full set of 37 dummy variables using `one_hot` option
 #' dummies_one_hot <- rec %>%
 #'   step_dummy(city, one_hot = TRUE) %>%
-#'   prep(training = Sacramento)
+#'   prep()
 #'
 #' dummy_data_one_hot <- bake(dummies_one_hot, new_data = NULL)
 #'
 #' dummy_data_one_hot %>%
 #'   select(starts_with("city")) %>%
-#'   names() # no reference level
+#'   glimpse() # no reference level
 #'
+#' # Obtain the full set of 37 dummy variables using helmert contrasts
+#' dummies_helmert <- rec %>%
+#'   step_dummy(city, contrasts = contr.helmert) %>%
+#'   prep()
+#'
+#' dummy_data_helmert <- bake(dummies_helmert, new_data = NULL)
+#'
+#' dummy_data_helmert %>%
+#'   select(starts_with("city")) %>%
+#'   glimpse() # no reference level
 #'
 #' tidy(dummies, number = 1)
 #' tidy(dummies_one_hot, number = 1)
+#' tidy(dummies_helmert, number = 1)
 step_dummy <-
   function(
     recipe,
     ...,
     role = "predictor",
     trained = FALSE,
     one_hot = FALSE,
+    contrasts = list(
+      unordered = stats::contr.treatment,
```

Comment:
I think that using the function value would attach additional environments to the object (e.g. packages and global). I'm ok with character values but maybe require a call instead (or convert these to a call?).

Reply:
done in https://github.com/tidymodels/recipes/pull/1484/commits/ad7d62d6321bc82265b79644381fd8253f62ce0e